### ISSUE TITLE
Eject option

### ISSUE THREAD

**Issue by alainjacomet** [marmelab/react-admin#1862]
**Is your feature request related to a problem? Please describe.**

I have first hand experience with people reluctant to use this library because of the closed system it provides. An ejection mechanism similar to `create-react-app`'s would solve a lot of the concerns regarding this being a closed system that signifies a potential risk.

**Describe the solution you'd like**

An `eject` npm command that replaces the `react-admin` dependency for all the sub-dependency tree, and exposes the files that are currently providing abstractions.

**Describe alternatives you've considered**

At the moment the only alternative is not to use `react-admin` at all.

**Additional context**

I am not sure how achievable or easy this it at the moment. If I can get some insights from the maintainers, I might be able to make a PR myself, provided that this is a desirable feature.

**Comment by fzaninotto** [comment]
I don't understand your demand, nor do I understand the problems of the people you mention.

Contrary to create-react-app, react-admin isn't a generator ; it's a framework. You could compare it to Symfony or React-native. Even if those libraries are assembling various modules themselves, once you start building an app with them, you can't "eject" them. 

Could you give more insights about what bothers you and the other people about react-admin? Could you rephrase what you mean by "eject" in the context of react-admin?

**Comment by alainjacomet** [comment]
`react-admin` provides abstractions for common configurations that the user must do themselves if they are not using the library. _Ejecting_ would mean to expose the source code of these abstractions into a directory inside the project, and replacing the `react-admin` dependency in package.json with the direct sub dependencies ( material-ui, redux, redux-form, redux-saga, react-router, recompose, reselect, etc.)

**Comment by fzaninotto** [comment]
I'm still not getting you. What abstractions are you referring to exactly? Can you give an example of what result you expect from an *eject*?

**Comment by fzaninotto** [comment]
Bump

**Comment by alainjacomet** [comment]
Looks like this wasn't as straightforward as I initially thought. Maybe someone else finds this feature attractive and reopens the issue down the road :+1:
